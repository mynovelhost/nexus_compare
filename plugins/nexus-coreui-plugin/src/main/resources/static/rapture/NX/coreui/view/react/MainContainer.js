/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Main Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.MainContainer', {
  extend: 'Ext.Component',
  alias: 'widget.nx-coreui-react-main-container',

  reactView: undefined,

  scrollable: true,

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    var reactElement = react.createElement(this.reactView);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
  }
});
