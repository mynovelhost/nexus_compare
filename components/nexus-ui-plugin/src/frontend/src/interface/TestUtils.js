/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {act} from 'react-dom/test-utils';
import {render} from '@testing-library/react';

/**
 * @since 3.24
 */
export default class Utils {
  static render(view, extraSelectors) {
    const selectors = render(view);
    const {queryByText} = selectors;
    return {
      ...selectors,
      loadingMask: () => queryByText('Loadingâ€¦'),
      ...extraSelectors(selectors)};
  }
}
