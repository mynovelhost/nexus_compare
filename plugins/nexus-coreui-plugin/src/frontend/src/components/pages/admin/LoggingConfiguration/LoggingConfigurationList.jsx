/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import {useMachine} from '@xstate/react';

import {faChevronRight, faPlusCircle, faRedo} from '@fortawesome/free-solid-svg-icons';

import {
  Button,
  NxFilterInput,
  NxFontAwesomeIcon,
  NxTable,
  NxTableBody,
  NxTableCell,
  NxTableHead,
  NxTableRow,
  Section,
  SectionActions,
  SectionHeader
} from 'nexus-ui-plugin';

import LoggingConfigurationListMachine, {ASC} from './LoggingConfigurationListMachine';

import UIStrings from '../../../../constants/UIStrings';

export default function LoggingConfigurationList({onCreate, onEdit}) {
  const [current, send] = useMachine(LoggingConfigurationListMachine, {devTools: true});
  const isLoading = current.matches('loading');
  const data = current.context.data;
  const filterText = current.context.filter;
  const error = current.context.error;

  function getSortDirection(fieldName, sortField, sortDirection) {
    if (sortField !== fieldName) {
      return null;
    }
    return sortDirection === ASC ? 'asc' : 'desc';
  }

  const nameSortDir = getSortDirection('name', current.context.sortField, current.context.sortDirection);
  const levelSortDir = getSortDirection('level', current.context.sortField, current.context.sortDirection);

  function filter(value) {
    send('FILTER', {filter: value});
  }

  function clearFilter() {
    send('FILTER', {filter: ''});
  }

  function reset() {
    send('RESET');
  }

  return <Section className="nxrm-logging-configuration-list">
    <SectionActions>
      <NxFilterInput
          inputId="filter"
          onChange={filter}
          onClear={clearFilter}
          value={filterText}
          placeholder={UIStrings.LOGGING.FILTER_PLACEHOLDER}/>
    </SectionActions>
    <SectionHeader>
      <Button variant="primary" onClick={onCreate}>
        <NxFontAwesomeIcon icon={faPlusCircle}/>
        <span>{UIStrings.LOGGING.CREATE_BUTTON}</span>
      </Button>
      <Button onClick={reset}>
        <NxFontAwesomeIcon icon={faRedo}/>
        <span>{UIStrings.LOGGING.RESET_ALL_BUTTON}</span>
      </Button>
    </SectionHeader>
    <NxTable>
      <NxTableHead>
        <NxTableRow>
          <NxTableCell onClick={() => send('SORT_BY_NAME')} isSortable sortDir={nameSortDir}>
            {UIStrings.LOGGING.NAME_LABEL}
          </NxTableCell>
          <NxTableCell onClick={() => send('SORT_BY_LEVEL')} isSortable sortDir={levelSortDir}>
            {UIStrings.LOGGING.LEVEL_LABEL}
          </NxTableCell>
          <NxTableCell hasIcon />
        </NxTableRow>
      </NxTableHead>
      <NxTableBody isLoading={isLoading} error={error}>
        {data.map(({name, level}) => (
            <NxTableRow key={name} onClick={() => onEdit(name)} isClickable>
              <NxTableCell>{name}</NxTableCell>
              <NxTableCell>{level}</NxTableCell>
              <NxTableCell hasIcon><NxFontAwesomeIcon icon={faChevronRight} /></NxTableCell>
            </NxTableRow>
        ))}
      </NxTableBody>
    </NxTable>
  </Section>;
}
