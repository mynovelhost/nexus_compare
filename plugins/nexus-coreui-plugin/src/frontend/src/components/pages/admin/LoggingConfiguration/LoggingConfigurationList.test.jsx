/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import {act} from 'react-dom/test-utils';
import {fireEvent, render, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TestUtils from 'nexus-ui-plugin/src/frontend/src/interface/TestUtils';
import axios from 'axios';

import LoggingConfigurationList from './LoggingConfigurationList';

import UIStrings from "../../../../constants/UIStrings";

jest.mock('axios', () => ({
  ...jest.requireActual('axios'), // Use most functions from actual axios
  get: jest.fn()
}));

describe('LoggingConfigurationList', function() {
  function renderView(view = <LoggingConfigurationList/>) {
    return TestUtils.render(view, ({queryByPlaceholderText}) => ({
      filter: () => queryByPlaceholderText(UIStrings.LOGGING.FILTER_PLACEHOLDER)
    }));
  }

  it('renders the resolved data', async function() {
    const rows = [
      {name: 'ROOT', level: 'INFO'},
      {name: 'org.sonatype.nexus', level: 'DEBUG'}
    ];

    axios.get.mockReturnValue(Promise.resolve({
      data: rows
    }));

    const {container, loadingMask} = renderView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    rows.forEach((row, i) => {
      expect(container.querySelector(`tbody tr:nth-child(${i+1}) td:nth-child(1)`)).toHaveTextContent(row.name);
      expect(container.querySelector(`tbody tr:nth-child(${i+1}) td:nth-child(2)`)).toHaveTextContent(row.level);
    });
    expect(container).toMatchSnapshot();
  });

  it('renders a loading spinner', async function() {
    axios.get.mockReturnValue(new Promise(() => {}));

    const {container, loadingMask} = renderView();

    expect(loadingMask()).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders an error message', async function() {
    axios.get.mockReturnValue(Promise.reject({message: 'Error'}));

    const {container, loadingMask} = renderView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    expect(container.querySelector('td.nx-error')).toHaveTextContent('Error');
    expect(container).toMatchSnapshot();
  });

  it('searches for lower-cased strings in the logger names', async function() {
    axios.get.mockReturnValue(Promise.resolve({
      data: [
        {name: 'ROOT', level: 'INFO'},
        {name: 'org.sonatype.nexus', level: 'DEBUG'}
      ]
    }));

    const {container, loadingMask, filter} = renderView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    fireEvent.change(filter(), {target: {value: 'sonatype'}});

    expect(container.querySelector('tbody tr:nth-child(1) td:nth-child(1)')).toHaveTextContent('org.sonatype.nexus');
  });
});
