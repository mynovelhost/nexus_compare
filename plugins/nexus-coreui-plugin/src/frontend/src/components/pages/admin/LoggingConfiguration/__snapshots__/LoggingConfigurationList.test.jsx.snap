// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggingConfigurationList renders a loading spinner 1`] = `
<div>
  <div
    class="nxrm-section nx-tile nxrm-logging-configuration-list"
  >
    <div
      class="nxrm-section-actions nx-tile__actions"
    >
      <div
        class="nx-filter-input"
      >
        <span
          class="nx-filter-input__add-on"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-filter fa-w-16 nx-icon"
            data-icon="filter"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          autocomplete="off"
          class="nx-filter-text-input"
          id="filter"
          placeholder="Filter by Logger Name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="nxrm-section-header nx-tile-header"
    >
      <button
        class="nx-btn nx-btn--primary"
        type="input"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus-circle fa-w-16 nx-icon"
          data-icon="plus-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
          />
        </svg>
        <span>
          Create Logger
        </span>
      </button>
      <button
        class="nx-btn"
        type="input"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-redo fa-w-16 nx-icon"
          data-icon="redo"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500.33 0h-47.41a12 12 0 0 0-12 12.57l4 82.76A247.42 247.42 0 0 0 256 8C119.34 8 7.9 119.53 8 256.19 8.1 393.07 119.1 504 256 504a247.1 247.1 0 0 0 166.18-63.91 12 12 0 0 0 .48-17.43l-34-34a12 12 0 0 0-16.38-.55A176 176 0 1 1 402.1 157.8l-101.53-4.87a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12h200.33a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12z"
            fill="currentColor"
          />
        </svg>
        <span>
          Reset to Default Levels
        </span>
      </button>
    </div>
    <table
      class="nx-table"
    >
      <thead>
        <tr
          class="nx-table-row nx-table-row--header"
        >
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Logger Name
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-down fa-w-10 nx-icon"
                data-icon="sort-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-up fa-w-10 nx-icon"
                data-icon="sort-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Logger Level
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--icon"
          />
        </tr>
      </thead>
      <tbody>
        <tr
          class="nx-table-row"
        >
          <td
            class="nx-cell nx-cell--empty"
            colspan="3"
          >
            <div
              class="nx-loading-spinner"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-notch fa-w-16 fa-spin nx-loading-spinner__icon nx-icon"
                data-icon="circle-notch"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Loadingâ€¦
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LoggingConfigurationList renders an error message 1`] = `
<div>
  <div
    class="nxrm-section nx-tile nxrm-logging-configuration-list"
  >
    <div
      class="nxrm-section-actions nx-tile__actions"
    >
      <div
        class="nx-filter-input"
      >
        <span
          class="nx-filter-input__add-on"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-filter fa-w-16 nx-icon"
            data-icon="filter"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          autocomplete="off"
          class="nx-filter-text-input"
          id="filter"
          placeholder="Filter by Logger Name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="nxrm-section-header nx-tile-header"
    >
      <button
        class="nx-btn nx-btn--primary"
        type="input"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus-circle fa-w-16 nx-icon"
          data-icon="plus-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
          />
        </svg>
        <span>
          Create Logger
        </span>
      </button>
      <button
        class="nx-btn"
        type="input"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-redo fa-w-16 nx-icon"
          data-icon="redo"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500.33 0h-47.41a12 12 0 0 0-12 12.57l4 82.76A247.42 247.42 0 0 0 256 8C119.34 8 7.9 119.53 8 256.19 8.1 393.07 119.1 504 256 504a247.1 247.1 0 0 0 166.18-63.91 12 12 0 0 0 .48-17.43l-34-34a12 12 0 0 0-16.38-.55A176 176 0 1 1 402.1 157.8l-101.53-4.87a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12h200.33a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12z"
            fill="currentColor"
          />
        </svg>
        <span>
          Reset to Default Levels
        </span>
      </button>
    </div>
    <table
      class="nx-table"
    >
      <thead>
        <tr
          class="nx-table-row nx-table-row--header"
        >
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Logger Name
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-down fa-w-10 nx-icon"
                data-icon="sort-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-up fa-w-10 nx-icon"
                data-icon="sort-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Logger Level
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--icon"
          />
        </tr>
      </thead>
      <tbody>
        <tr
          class="nx-table-row"
        >
          <td
            class="nx-cell nx-error"
            colspan="3"
          >
            <span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-exclamation-triangle fa-w-18 fa-fw nx-icon"
                data-icon="exclamation-triangle"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                  fill="currentColor"
                />
              </svg>
              Error
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LoggingConfigurationList renders the resolved data 1`] = `
<div>
  <div
    class="nxrm-section nx-tile nxrm-logging-configuration-list"
  >
    <div
      class="nxrm-section-actions nx-tile__actions"
    >
      <div
        class="nx-filter-input"
      >
        <span
          class="nx-filter-input__add-on"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-filter fa-w-16 nx-icon"
            data-icon="filter"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          autocomplete="off"
          class="nx-filter-text-input"
          id="filter"
          placeholder="Filter by Logger Name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="nxrm-section-header nx-tile-header"
    >
      <button
        class="nx-btn nx-btn--primary"
        type="input"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus-circle fa-w-16 nx-icon"
          data-icon="plus-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
          />
        </svg>
        <span>
          Create Logger
        </span>
      </button>
      <button
        class="nx-btn"
        type="input"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-redo fa-w-16 nx-icon"
          data-icon="redo"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500.33 0h-47.41a12 12 0 0 0-12 12.57l4 82.76A247.42 247.42 0 0 0 256 8C119.34 8 7.9 119.53 8 256.19 8.1 393.07 119.1 504 256 504a247.1 247.1 0 0 0 166.18-63.91 12 12 0 0 0 .48-17.43l-34-34a12 12 0 0 0-16.38-.55A176 176 0 1 1 402.1 157.8l-101.53-4.87a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12h200.33a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12z"
            fill="currentColor"
          />
        </svg>
        <span>
          Reset to Default Levels
        </span>
      </button>
    </div>
    <table
      class="nx-table"
    >
      <thead>
        <tr
          class="nx-table-row nx-table-row--header"
        >
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Logger Name
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-down fa-w-10 nx-icon"
                data-icon="sort-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-up fa-w-10 nx-icon"
                data-icon="sort-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Logger Level
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--icon"
          />
        </tr>
      </thead>
      <tbody>
        <tr
          class="nx-table-row nx-clickable"
        >
          <td
            class="nx-cell"
          >
            <span>
              ROOT
            </span>
          </td>
          <td
            class="nx-cell"
          >
            <span>
              INFO
            </span>
          </td>
          <td
            class="nx-cell nx-cell--icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-chevron-right fa-w-10 nx-icon"
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="nx-table-row nx-clickable"
        >
          <td
            class="nx-cell"
          >
            <span>
              org.sonatype.nexus
            </span>
          </td>
          <td
            class="nx-cell"
          >
            <span>
              DEBUG
            </span>
          </td>
          <td
            class="nx-cell nx-cell--icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-chevron-right fa-w-10 nx-icon"
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
